### DISCUSSION TITLE
How can I install code-server in vbox?

### DISCUSSION THREAD

**Discussion by cathaysia** [coder/code-server#2006] [Q&A] [answered]
- Web Browser: Firefox
- Local OS: Win
- Remote OS: Fedora
- Remote Architecture: x86_64
- `code-server --version`: 3.34

Hi, I want install code-server to fedora 32 server of vbox. However, when I start code-server service, I can't access it. And this [guide](https://github.com/cdr/code-server/blob/v3.4.1/doc/guide.md) seem useless for me.

My vbox use NAT(not NAT NETWORK), I forward :8080 to localhost:18080 by vbox, my web browser can't get any page and stop quickly. When I use ssh connect to fedora, and try `wget localhost:8080`, it could download a index.html.

Hava any way for me to resolve this problem?

![image](https://user-images.githubusercontent.com/42114817/91311644-e5f83f00-e7e5-11ea-94e4-7f35d3c89890.png)
![image](https://user-images.githubusercontent.com/42114817/91311698-f5778800-e7e5-11ea-9675-7f7a71761663.png)
![image](https://user-images.githubusercontent.com/42114817/91311728-ff00f000-e7e5-11ea-8cad-68b8bdcaee90.png)

**Comment by cathaysia** [comment, marked as answer]
Maybe I had slove this problem, I find this sentence **'It will listen onlocalhost to avoid exposing itself to the world'** in [Setup Guide#3](https://github.com/cdr/code-server/blob/v3.4.1/doc/guide.md#3-expose-code-server). So I guess this problem may cause by this address. I let code-server listen 0.0.0.0:8080, then forward 8080 to 18080 by vbox. Now it works!

This is my conf:
```nginx
# ~/.config/code-server
bind-addr: 0.0.0.0:8080
auth: password
password: 1234
cert: false
```

**Comment by bpmct** [comment]
Awesome! Glad you got it working :)
